src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetLinkingDataverseCommand.java

```diff
@@ -31,17 +31,19 @@ public class DeleteDatasetLinkingDataverseCommand extends AbstractCommand<Datase
     private final DatasetLinkingDataverse doomed;
     private final Dataset editedDs;
     private final boolean index;
+    private final boolean preAuthorized;
     
-    public DeleteDatasetLinkingDataverseCommand(DataverseRequest aRequest, Dataset editedDs , DatasetLinkingDataverse doomed, boolean index) {
+    public DeleteDatasetLinkingDataverseCommand(DataverseRequest aRequest, Dataset editedDs , DatasetLinkingDataverse doomed, boolean index, boolean preAuthorized) {
```

Comment:
I'm confused by this "preAuthorized" boolean. Shouldn't we be checking for the appropriate permission? Possibly that permission is "PublishDataset" based on what I'm seeing in the issue, but I'm not sure.

Reply:
I renamed this hasPermission. The way it works is: hasPermission = false is what is passed when the admin api is called since we don't have enough user information. This goes through the older code where the user has to be a superuser. Any call from the UI has already had the permissions checked when getting the list of linked datasets. Only datasets where the user has permission will be listed in the dropdown. Sending hasPermission = true allows us to carry out the unlink without needing to re-check all the permissions.

Reply:
I dunno. It feel like we have a nice system where commands are annotated with required permissions and this boolean is bypassing the system. Can you please see what @scolapasta thinks?

Reply:
The current command doesn't check permissions. It only looks for the superuser. By adding the permission check it would mean checking permissions twice, once when building the list, and second when the command executes.
@scolapasta

Reply:
Another way to think about this is to lean on our newish API-first approach.

Some day the SPA will need the ability to unlink datasets as non-superuser to match what this pull request is trying to deliver for the old JSF GUI. (In the issue, the user is arguing, "If I can link a dataset, I should be able to unlink it.") Therefore it would be ideal to improve the permission handling in the commands themselves so that the logic can be used by the API (and ultimately, the SPA).

I hope this helps. I don't know how much effort it is to improve the permission handling in the commands.

Reply:
@stevenwinship @pdurbin I had a chance to look this over, and while I understand it's more work and means the permissions get checked twice, I agree with @pdurbin that the command should follow the pattern of using the annotations for permission check here.

Reply:
I removed the hasPermission and superuser check. Using the @RequiredPermissions( Permission.PublishDataset )